pkg/virt-operator/util/config.go

```diff
@@ -214,7 +222,10 @@ func getConfig(registry, tag, namespace string, additionalProperties map[string]
 		}
 	}
 
-	config := newDeploymentConfigWithTag(registry, imagePrefix, tag, namespace, additionalProperties)
+	kubernetesServiceHost := os.Getenv(KubernetesServiceHost)
```

Comment:
These `KUBERNETES_SERVICE_PORT` and `KUBERNETES_SERVICE_HOST` env vars are always going to be present I believe. They're injected automatically regardless if we explicitly set them or not on a deployment.

I feel kind of weird saying we're going to take whatever k8s injected env vars virt-operator are given, and ultimately use those as overrides for every pod related to kubevirt. My feeling is that if we include some sort of behavior like this, it should be an opt-in sort of thing.

Reply:
They're injected at the Kubelet level, not on the containers themselves in the API, you won't see these environment variables if you query for the deployments unless they were explicitly set. Therefore it seems to make sense to pass these connection details on if they were explicitly provided by the deployer of the virt operator.

 See Kubelet code here: https://github.com/kubernetes/kubernetes/blob/7d1488f0f28c51da72bd50f8314767f47d0013eb/pkg/kubelet/kubelet_pods.go#L500

Reply:
> Therefore it seems to make sense to pass these connection details on if they were explicitly provided by the deployer of the virt operator.

this is where maybe I'm missing something. For most clusters, these two env-vars are going to be present within the container's environment already.  So, by default, we're going to always be propagating the KUBERNETES_* env vars ourselves whenever virt-operator sees those vars in it's environment (which it always will).

Does that make sense what I'm saying? I'll make a quick comment on the exact line that made this apparent to me.